web/frontend/src/components/encoder.vue

```diff
@@ -5,102 +5,101 @@ import { grpc } from '@improbable-eng/grpc-web';
 import { Client, encoderApi, ResponseStream, robotApi, type ServiceError } from '@viamrobotics/sdk';
 import { displayError } from '../lib/error';
 import { rcLogConditionally } from '../lib/log';
+import { scheduleAsyncPoll } from '../lib/schedule';
 
 const props = defineProps<{
   name: string
   client: Client
   statusStream: ResponseStream<robotApi.StreamStatusResponse> | null
 }>();
 
-let properties = $ref<encoderApi.GetPropertiesResponse.AsObject | undefined>();
+let properties = $ref<encoderApi.GetPropertiesResponse.AsObject>();
 let positionTicks = $ref(0);
 let positionDegrees = $ref(0);
 
-let refreshId = -1;
+let cancelPoll: (() => void) | undefined;
 
-const refresh = () => {
-  const req = new encoderApi.GetPositionRequest();
-  req.setName(props.name);
+const getProperties = () => new Promise<encoderApi.GetPropertiesResponse.AsObject>((resolve, reject) => {
+  const request = new encoderApi.GetPropertiesRequest();
+  request.setName(props.name);
 
-  rcLogConditionally(req);
-  props.client.encoderService.getPosition(
-    req,
-    new grpc.Metadata(),
-    (error: ServiceError | null, resp: encoderApi.GetPositionResponse | null) => {
-      if (error) {
-        return displayError(error as ServiceError);
-      }
+  rcLogConditionally(request);
+  props.client.encoderService.getProperties(request, new grpc.Metadata(), (error, response) => {
```

Comment:
Seems like a good candidate for a promisification helper in a future PR

Reply:
Yeah, I think as a next step I'd like to start moving these out into an `api` folder that promisifies all these callback apis. Quarantine zones FTW!